core/includes/form.inc

```diff
@@ -4594,7 +4596,13 @@ function form_validate_number(&$element, &$form_state) {
     $offset = isset($element['#min']) ? $element['#min'] : 0.0;
 
     if (!valid_number_step($value, $element['#step'], $offset)) {
-      form_error($element, t('%name is not a valid number.', array('%name' => $name)));
+      // Integer steps are the most common use case, so give a specific message.
+      if ($element['#step'] == 1 && is_numeric($offset) && $offset == round($offset)) {
+        form_error($element, t('%name: the value must be an integer.', array('%name' => $name)));
+      }
+      else {
+        form_error($element, t('%name: the value must be a whole number of steps %step from %offset.', array('%name' => $name, '%step' => $element['#step'], '%offset' => $offset)));
```

Comment:
Would something like this make better sense?:

```suggestion
        form_error($element, t('%name: the value must be a whole number of steps of %step, starting from %offset.', array('%name' => $name, '%step' => $element['#step'], '%offset' => $offset)));
```

Reply:
That is indeed better; better still (and what I changed to) is:

> the value must be a whole number of steps of size %step, starting from %offset.

Also just noticed the "integer" in line 4601. For consistency, this, too, should be "a whole number," so also changed.